client/src/components/MistakeForm.vue

```diff
@@ -161,6 +197,14 @@ export default Vue.extend({
       await this.$store.dispatch(MistakesActions.AddMistake, this.mistake);
       await this.$router.push('/journal/mistakes');
     },
+    async update(): Promise<void> {
+      this.mistakeEdit.name = this.mistake.name;
+      this.mistakeEdit.goal = this.mistake.goal;
+      this.mistakeEdit.tips = this.mistake.tips;
+      this.mistakeEdit.priority = this.mistake.priority;
```

Comment:
You cannot mutate computed property. Plus I don't think that is correct technically. You created action `UpdateMistake` and it should have eventually a mutation that sets mistake to a new one (updated one). You can send in this action `this.mistake`.